clippy_lints/src/operators/float_cmp.rs

```diff
@@ -18,8 +18,17 @@ pub(crate) fn check<'tcx>(
     left: &'tcx Expr<'_>,
     right: &'tcx Expr<'_>,
 ) {
-    if (op == BinOpKind::Eq || op == BinOpKind::Ne)
+    let peel_expr = |e: &'tcx Expr<'tcx>| match e.kind {
+        ExprKind::Cast(e, _) | ExprKind::AddrOf(BorrowKind::Ref, _, e) | ExprKind::Unary(UnOp::Neg, e) => Some(e),
+        _ => None,
+    };
+
+    if matches!(op, BinOpKind::Eq | BinOpKind::Ne)
+        && let left = peel_hir_expr_while(left, peel_expr)
+        && let right = peel_hir_expr_while(right, peel_expr)
         && is_float(cx, left)
```

Comment:
What do you think about:

```suggestion
    if !matches(op, BinOpKind::Eq | BinOpKind::Ne) {
        return;
    }
    let left = peel_hir_expr_while(left, peel_expr);
    let right = peel_hir_expr_while(right, peel_expr);
    if is_float(cx, left)
```

?

Reply:
Let chains with infallible bindings are used all over Clippy.

Reply:
I agree with @Jarcho, this is the normal way and readable, IMO.

Reply:
@Jarcho, if you rebase I can review this PR sometime this week. If you say it's ready